sdk/eventgrid/azure-eventgrid/samples/publish_samples/publish_with_shared_access_signature_sample.py

```diff
@@ -0,0 +1,54 @@
+import sys
+import os
+import json
+from random import randint, sample
+from typing import Sequence
+import time
+
+PACKAGE_PARENT = '..'
+SCRIPT_DIR = os.path.dirname(os.path.realpath(os.path.join(os.getcwd(), os.path.expanduser(__file__))))
+sys.path.append(os.path.normpath(os.path.join(SCRIPT_DIR, PACKAGE_PARENT)))
+
+from dateutil.tz import tzutc
+from datetime import timedelta, timezone
+import datetime as dt
+
+from azure.eventgrid import EventGridSharedAccessSignatureCredential
+from azure.mgmt.eventgrid import EventGridManagementClient
+from azure.eventgrid import EventGridPublisherClient, EventGridEvent, CloudEvent, generate_shared_access_signature
+from azure.core.exceptions import (
+    ResourceNotFoundError,
+    ResourceExistsError,
+    ClientAuthenticationError
+)
+
+key = os.environ.get("CUSTOM_ACCESS_KEY")
+topic_hostname = "eventgridcloudeventsub.eastus-1.eventgrid.azure.net"
```

Comment:
should base this (and other "caller-defined variables") off env-vars like you do key.